pkg/templates/machinecontroller/machines.go

```diff
@@ -96,6 +87,12 @@ func createMachineDeployment(cluster *kubeoneapi.KubeOneCluster, workerset kubeo
 		"workerset": workerset.Name,
 	}
 
+	if workerset.Config.Network != nil {
+		// we have static network config
```

Comment:
static network config is kinda special care and we need to flip those `maxSurge` and `maxUnavailable`